rest/src/main/java/at/jku/isse/ecco/rest/RepositoryService.java

```diff
@@ -0,0 +1,264 @@
+        return repositories.get(rId).variantAddFeature(variantId, featureId);
+    }
+
+    public RestRepository variantUpdateFeature(int rId, String variantId, String featureName, String id) {
+        return repositories.get(rId).variantUpdateFeature(variantId, featureName, id);
+    }
+
+    public RestRepository variantRemoveFeature(int rId, String variantId, String featureName) {
+        return repositories.get(rId).variantRemoveFeature(variantId, featureName);
+    }
+
+    public Path checkout(final int rId, final String variantId) {
+        Path checkoutFolder = repositories.get(rId).getPath().resolve("checkout");
+        Path checkoutZip = checkoutFolder.getParent().resolve("checkout.zip");
+
+        if(checkoutFolder.toFile().exists()){       //delete old checkout folder
+            if(!deleteDirectory(checkoutFolder.toFile())) {
+                throw new HttpStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Deleting old directory is not possible: " + checkoutFolder.getParent().getFileName().toString());
+            }
+        }
+        if(checkoutZip.toFile().exists()){      //delete old checkout zip file
+            if(!deleteDirectory(checkoutZip.toFile())) {
+                throw new HttpStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Deleting old zip-file is not possible: " + checkoutZip.getParent().getFileName().toString());
+            }
+        }
+
+        if(!checkoutFolder.toFile().mkdir()) {      //create new checkout folder
+            throw new HttpStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Creation failed: " + checkoutZip.getParent().getFileName().toString());
+        }
+
+        repositories.get(rId).checkout(variantId, checkoutFolder);      //handler checkout
+
+        try {
+            zipFolder(checkoutFolder, checkoutZip);
+        } catch (Exception e) {
+            throw new HttpStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Failed to create checkout zip file");
+        }
+
+        LOGGER.info(rId + ": checked out");
+        return checkoutZip;
+    }
+
+    // Feature ---------------------------------------------------------------------------------------------------------
+    public RestRepository setFeatureDescription(int rId, String featureId, String description) {
+        return repositories.get(rId).setFeatureDescription(featureId, description);
+    }
+
+    public RestRepository setFeatureRevisionDescription(int rId, String featureId, String revisionId, String description) {
+        return repositories.get(rId).setFeatureRevisionDescription(featureId, revisionId, description);
+    }
+
+    public void pullFeaturesRepository(final int toRId, final int oldRId, final String deselectedFeatures) {
+        repositories.get(toRId).fork(repositories.get(oldRId), deselectedFeatures);     //handler fork
+    }
+
+    // private methods -------------------------------------------------------------------------------------------------
```

Comment:
unnecessary comment, private methods are already described by the `private` keyword.